File: client/goood-back/src/services/entries/Feedback.ts
export class Feedback {
    id?: string
    message!: string
    creation_date!: string
    type!: number

    public emoji (): string {
        switch (this.type) {
            case 0: return '👍'
            case 1: return '👎'
            case 2: return '🔥'
            case 3: return '🤬'
        }

        return '😨'
    }
}

File: server/api/feedback/entries/Feedback.go
package entries

import "time"

type Feedback struct {
	NewFeedback
	CreationDate time.Time `json:"creation_date"`
}

type FeedbackSortable []Feedback

func (f FeedbackSortable) Len() int {
	return len(f)
}

func (f FeedbackSortable) Less(i, j int) bool {
	return f[i].CreationDate.After(f[j].CreationDate)
}

func (f FeedbackSortable) Swap(i, j int) {
	f[i], f[j] = f[j], f[i]
}
